core/trino-main/src/test/java/io/trino/sql/planner/TestLogicalPlanner.java

```diff
@@ -965,6 +965,36 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                         node(ValuesNode.class)))))));
     }
 
+    @Test
+    public void testCorrelatedDistinctAggregationRewriteToLeftOuterJoin()
+    {
+        assertPlan(
```

Comment:
so this is testing `PushAggregationThroughOuterJoin`?

Reply:
Yes. It checks the de-correlation rule `TransformCorrelatedGlobalAggregationWithoutProjection` combined with `PushAggregationThroughOuterJoin`. Before the change, `PushAggregationThroughOuterJoin` didn't fire because of misplaced "distinct operator".